packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart

```diff
@@ -38,18 +40,35 @@ class TimelineScreenBody extends StatefulWidget {
 }
 
 class TimelineScreenBodyState extends State<TimelineScreenBody> {
-  @visibleForTesting
-  final controller = TimelineController();
+  TimelineController controller;
+
+  StreamSubscription selectedFrameSubscription;
 
   @override
   void initState() {
     super.initState();
   }
 
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    controller = Controllers.of(context).timeline;
+    controller.timelineService.updateListeningState(true);
+
+    // TODO(kenz): use Notifier class to register and unregister listeners.
+    selectedFrameSubscription?.cancel();
+    selectedFrameSubscription =
+        controller.frameBasedTimeline.onSelectedFrame.listen((_) {
+      setState(() {});
+    });
```

Comment:
add AutoDisposeMixin to TimelineScreenBodyState and then here write:
```
autoDispose(_controller.frameBasedTimeline.onSelectedFrame.listen((_) {
      setState(() {});
    }));
```
Then you can remove the selectedFrameSubscription var and the calls to cancel.

Reply:
A bit leery to make this change before I head out.

Reply:
okay no worries. Can you add a TODO? I can clean up when you're out next week 👍